src/dispatch/plugins/dispatch_slack/views.py

```diff
@@ -681,6 +691,96 @@ def get_channel_id(event_body: dict):
     return channel_id
 
 
+def create_incident_open_modal(command: dict = None, db_session=None):
+    """
+    Prepare the Modal / View
+    Ask slack to open a modal with the prepared Modal / View content
+    """
+    channel_id = command.get("channel_id")
+    trigger_id = command.get("trigger_id")
+
+    type_options = []
+    for t in incident_type_service.get_all(db_session=db_session):
+        type_options.append({"label": t.name, "value": t.name})
+
+    priority_options = []
+    for priority in incident_priority_service.get_all(db_session=db_session):
+        priority_options.append({"label": priority.name, "value": priority.name})
+
+    modal_view_template = create_modal_content(
+        channel_id=channel_id,
+        incident_types=type_options,
+        incident_priorities=priority_options
+    )
+
+    dispatch_slack_service.open_view_for_user(client=slack_client, trigger_id=trigger_id, view=modal_view_template)
+
+
+def parse_submitted_form(view_data: dict):
+    """Parse the submitted data and return important / required fields for dispatch to start an incident"""
+    parsed_data = {}
+    state_elem = view_data.get('state')
+    state_values = state_elem.get('values')
+    for state in state_values:
+        state_key_value_pair = state_values[state]
+
+        for elem_key in state_key_value_pair:
+            elem_key_value_pair = state_values[state][elem_key]
+
+            if elem_key_value_pair.get('selected_option') and elem_key_value_pair.get('selected_option').get('value'):
+                parsed_data[state] = {
+                    'name': elem_key_value_pair.get('selected_option').get('text').get('text'),
+                    'value': elem_key_value_pair.get('selected_option').get('value')
+                }
+            else:
+                parsed_data[state] = elem_key_value_pair.get('value')
+
+    return parsed_data
+
+
+def create_incident_from_submitted_form(
+        user_id: str, user_email: str, incident_id: int, action: dict, db_session=None
+):
+    from dispatch.incident.service import create
+    from dispatch.incident.flows import incident_create_flow
+    from dispatch.incident.enums import IncidentSlackViewBlockId, IncidentStatus
```

Comment:
You don't need these imports. You can do `incident_service.create()`, `incident_flows. incident_create_flow()`, and use the enums already imported.